server/publications/collections/products.js

```diff
@@ -320,24 +324,34 @@ Meteor.publish("Products", function (productScrollLimit = 24, productFilters, so
     return this.ready();
   }
 
-  // Get active shop id's to use for filtering
-  const activeShopsIds = Shops.find({
-    $or: [
-      { "workflow.status": "active" },
-      { _id: primaryShopId }
-    ]
-  }, {
-    fields: {
-      _id: 1
-    }
-  }).fetch().map((activeShop) => activeShop._id);
+  // if the current shop is the primary shop, get products from all shops
+  // otherwise, only list products from _this_ shop.
+  if (shopId === primaryShopId) {
+    activeShopsIds = Shops.find({
+      $or: [
+        { "workflow.status": "active" },
+        { _id: primaryShopId }
+      ]
+    }, {
+      fields: {
+        _id: 1
+      }
+    }).fetch().map((activeShop) => activeShop._id);
+  } else {
+    activeShopsIds = [shopId];
+  }
 
   const selector = filterProducts(productFilters);
 
   if (selector === false) {
     return this.ready();
   }
 
+  // if shops were requested in the product filter
+  if (selector.shopId && Array.isArray(selector.shopId.$in)) {
+    activeShopsIds = _.intersection(selector.shopId.$in, activeShopsIds);
```

Comment:
product filters allow for a set of shops.
While some work had been done to validate the requested Shops, we wound up filtering with `{ $in: activeShopIds }`, discarding the requested shops. this restores that behavior